File: src/main/resources/com/zkdas/oop/service/IdGenerator.cs
namespace OOP.service
{
    /// <summary>
    ///  класс для генерации id
    /// </summary>
    public class IdGenerator
    {
        private int _lastId;

        public IdGenerator()
        {
            _lastId = 0;
        }

        public int get_next_id()
        {
            _lastId++;
            return _lastId;
        }
    }
}

File: src/main/java/com/zkdas/oop/service/IdGenerator.java
package com.zkdas.oop.service;

public class IdGenerator {
    /**
     * Класс для генерации id
     */
    private int _lastId; // последний занятый id

    /**
     * Конструктор (инициализирует поле _lastId)
     */
    public IdGenerator() {
        _lastId = 0;
    }

    /**
     * Вернет следующий id
     * @return int число
     */
    public int get_next_id() {
        _lastId++;
        return _lastId;
    }
}
